src/git.ts

```diff
@@ -9,14 +9,17 @@ import {
 
 export const commitChangesFromRepo = async ({
   base,
-  repoDirectory = process.cwd(),
+  repoDirectory,
+  addDirectory,
```

Comment:
I'd recommend a more flexible `filter` that could be a glob or similar

Reply:
Given that we need to specify which directory to start from when performing a walk to determine which files to include, allowing users to specify this would be the most efficient way of emulating `git add .`, which is the use-case for `changesets/action`. I think we should still support specifying this parameter and use it in `changesets/action`, but I'm happy to also add a `filter` as it's not too much additional work.

Reply:
Hmm looks like I'm mis-remembering what the `dir` option in walk() does, and it needs to match the repo, so i'll have to implement custom filtering logic for `addFromDirectory` anyway, but given the simpler interface of `addFromDirectory` i'll keep the interface the same.